### ISSUE TITLE
Uninstalling chef-client does not remove chef service.

### ISSUE THREAD

**Issue by vinyar** [chef/chef#3825]
**OS:**
Windows 2012r2

**Scenario:**
Chef client 11.16.4 is installed
Chef client is configured to run as a service via chef-client cookbook.
Chef client 12.4.1 is installed on top of existing chef-client installation. Service option is not selected.
Chef client 12.4.1 is uninstalled.
chef-client service is left behind in the services.msc

**Error:**
`chef-client service is left behind in the services.msc`

**Expectation:**
chef-client is removed when chef-client is uninstalled regardless of the version.

@gallopinto , @tutocr

**Comment by coderanger** [comment]
Chef's packages don't include any service configuration, this was probably created by the chef-client cookbook or similar.

**Comment by vinyar** [comment]
bug updated with OS info

related:
https://github.com/chef/chef/issues/3826
https://github.com/chef/chef/issues/3824

**Comment by coderanger** [comment]
Ahh, windows. We use whatever is built in to WiX for the service stuffs on that platform. Specifically https://github.com/chef/omnibus-chef/blob/master/resources/chef/msi/source.wxs.erb#L62-L80

I defer to people that know WiX better (/cc @adamedx @schisamo) as to if this is something we can fix but my guess would be no.

**Comment by adamedx** [comment]
I think this will actually work if the msi feature for the service is used to install Chef, rather than having chef-service-manager.bat call directly into Ruby for this. Then MSI will know that it configured the service.

Another option is to simply make the msi uninstall the service regardless -- that has some risks.

@smurawski's solution is to do away with the service altogether and implement it as a scheduled task...

**Comment by vinyar** [comment]
I actually don't know benefits to having chef-client as a service. It's not a listening api.

Administratively, scheduled task is much easier, as compared to windows services which are a black box.

**Comment by smurawski** [comment]
Created epic https://github.com/chef/chef/issues/4042 to track this

**Comment by btm** [comment]
I would argue against the MSI uninstalling the service if it didn't install it.

We will investigate and confirm if the MSI will uninstall the service if it did install it.

Ultimately, we're recommending that people moving away from the use of the service. It was removed from the chef-client cookbook but then added back in https://github.com/chef-cookbooks/chef-client/commit/a93379f971fdaf68235fce7fdd4d56544162e8f1. So maybe it is a moot point regarding if the MSI removes a chef-client service that it did not create or not?

**Comment by NimishaS** [comment]
I had the following observation:
1. Installing chef-client using MSI with `ChefServiceFeature` option sets the `chef-client` service, but it's status is `Stopped`.
2. Uninstalling the MSI removes the service.

**Comment by btm** [comment]
I spent some time thinking about the state of the chef-client service being stopped, and I realize this is the right know.

You're usually (always?) going to want this order:
1. Install chef-client
2. Configure chef-client (client.rb, keys, etc)
3. Run chef-client
4. Start chef-client service

knife bootstrap does 1-3. Maybe most users don't notice 4 because part of their initial bootstrap includes a reboot, and since the service is configured as auto, it would start after a reboot. Otherwise you'd need to make sure the service was running in a recipe.

Uninstalling the chef-client MSI doesn't undo anything else that you might have done to the system using Chef, so I don't think we want to have behavior where we try to guess that you want us to uninstall the service created by a cookbook. Maybe you're upgrading, in which case there's no need to,

I understand that it would nice for some users if it did, but I think if we tried to fix the behavior that @vinyar described, we're very likely to cause unintended behavior for other users.
